### PR TITLE
Change bitmap loader

### PR DESCRIPTION
**Pull request by eeue56**
Not ready for review yet, opening to track a bit easier

### CODE REVIEW

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+				return Color.FromArgb((uint)color);
+			}
+			else if defined(CIL)
+			{
+				var color = NativeBitmap.GetPixel(x, y);
+				if (IsCMYK(NativeBitmap)){
+					debug_log "Color: ";
+					var thing = FromCmyk(color);
+
+					debug_log "returning " + thing;
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: This here seems like left-over debug-code...

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 2]
```diff
@@ -303,6 +304,7 @@ namespace Fuse.Internal.Bitmaps
 	{
 		public void Dispose()
 		{
+			debug_log "dispose coming in..";
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: You probably didn't mean to commit this, did you?

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 3]
```diff
@@ -398,7 +408,7 @@ namespace Fuse.Internal.Bitmaps
 			if (bundleFileSource != null)
 				return LoadFromBundleFile(bundleFileSource.BundleFile);
 
-
+			debug_log "File source" + fileSource.Name;
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: ...nor this

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+			return color.Z | (color.Y << 8) | (color.X << 16) | (color.W << 24); // encode as 0xAARRGGBB
+		@}
+	}
+
+	[extern(CPLUSPLUS) Require("Header.Include", "uImage/Bitmap.h")]
+	public sealed class Bitmap : IDisposable
+	{
+		public void Dispose()
+		{
+			debug_log "dispose coming in..";
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: Don't add random `debug_log`s please

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+			NativeBitmap = nativeBitmap;
+			_size = CPlusPlusHelpers.GetSize(nativeBitmap);
+		}
+
+		static Bitmap LoadFromBundleFile(BundleFile bundleFile)
+		{
+			if (bundleFile.BundlePath.EndsWith(".gif"))
+			{
+				// TODO: Fuse.Diagnostic.InternalError
+				debug_log "Gif files are not fully supported!";
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: debug_log is probably not what you want here.
Either ` Uno.Diagnostics.Debug.Log()`, or `Fuse.Diagnostics.UserWarning()`  is going to be more appropriate.

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+		}
+
+		public static Bitmap LoadFromFileSource(FileSource fileSource)
+		{
+			var bundleFileSource = fileSource as BundleFileSource;
+			if (bundleFileSource != null)
+				return LoadFromBundleFile(bundleFileSource.BundleFile);
+
+			debug_log "File source" + fileSource.Name;
+			debug_log "Inside the loader..";
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: This shouldn't be commited, I think

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+
+		static Bitmap LoadFromBundleFile(BundleFile bundleFile)
+		{
+			if (bundleFile.BundlePath.EndsWith(".gif"))
+			{
+				// TODO: Fuse.Diagnostic.InternalError
+				debug_log "Gif files are not fully supported!";
+			}
+
+			debug_log "File path:" + bundleFile.BundlePath;
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: Drop this

File: Source/Fuse.Common/Internal/Bitmap.uno (hunk 6 above)
Comment (eeue56) [inline Source/Fuse.Common/Internal/Bitmap.uno]: @kusma This is still WIP, I committed to move debug traces between machines so I can test on another platform (force-pushed with the `debug_log`s gone away)

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 8]
```diff
@@ -8,13 +8,25 @@ using Fuse.Platform;
 
 namespace System.Drawing
 {
+	[DotNetType("System.Array")]
+ public extern(CIL) class CilArray
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: Broken indentation

Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: Also, it feels wrong to put System.Array under the System.Drawing namespace...

File: Source/Fuse.Common/Internal/Bitmap.uno [hunk 9]
```diff
@@ -8,13 +8,25 @@ using Fuse.Platform;
...
+	{
+	}
+
+	[DotNetType("System.Runtime.InteropServices.Marshal")]
+	public extern(CIL) static class Marshal
```
Comment (kusma) [inline Source/Fuse.Common/Internal/Bitmap.uno]: This also doesn't feel like it belongs here...

File: Source/Fuse.Elements/Resources/FileImageSource.uno [hunk 10]
```diff
@@ -193,8 +194,11 @@ namespace Fuse.Resources
 					@}
 				}
 
-				var data = _file.ReadAllBytes();
-				TextureLoader.ByteArrayToTexture2DFilename(new Buffer(data), _file.Name, SetTexture);
+				var bitmap = Bitmap.LoadFromFileSource(_file);
+				var texture = bitmap.UploadTexture();
+				SetTexture(texture);
+				//var data = _file.ReadAllBytes();
+				//TextureLoader.ByteArrayToTexture2DFilename(new Buffer(data), _file.Name, SetTexture);
```
Comment (kusma) [inline Source/Fuse.Elements/Resources/FileImageSource.uno]: You should probably delete the commented-out code here...

File: Source/Fuse.Elements/Resources/FileImageSource.uno [hunk 11]
```diff
@@ -251,8 +257,9 @@ namespace Fuse.Resources
 			{
 				try
 				{
-					var data = _file.ReadAllBytes();
```
Comment (kusma) [inline Source/Fuse.Elements/Resources/FileImageSource.uno]: This moves the IO to the calling-thread, doesn't it? If so, I think this is a bad move. Can't we just wrap this in a buffer here instead, like the old code did?

**Comment by mortoray** [comment]
Closing as old. I don't see a particular issue this was addressing, so no issue has been created.

**Pull request closed without merging**